[IMP] website: update form protection page (#17298)
odoo/documentation: content/applications/websites/website/configuration/spam_protection.rst

Reviewer: I think I would add this as part of the steps below and rephrase?

```suggestion
#. `Create  <https://accounts.google.com/signup>`_ or `sign in <https://accounts.google.com/signin>`_ to a Google account.
#. <Open `the reCAPTCHA website registration page <https://www.google.com/recaptcha/admin/create>`_.
```

submissions using non-interactive challenges based on telemetry and visitor behavior.

.. important::
   We recommend using **Cloudflare Turnstile**, as reCAPTCHA v3 may not be compliant with local data
   protection regulations.

.. note::
   All pages using the :guilabel:`Form`, :guilabel:`Newsletter Block`, or :guilabel:`Newsletter

>    Popup` snippets are protected by both tools.

Reviewer: the [task](https://www.odoo.com/odoo/project.task/5409890) also mentions the login page, should we add it here as well?

On Cloudflare
-------------

#. `Create <https://dash.cloudflare.com/sign-up>`_ a Cloudflare account or use an existing one and

>    `log in <https://dash.cloudflare.com/login>`_.

Reviewer: ```suggestion
#. `Create <https://dash.cloudflare.com/sign-up>`_ or `sign in <https://dash.cloudflare.com/login>`_ to a Cloudflare acount.
```

#. On the dashboard's navigation sidebar, open the :guilabel:`Application security` drop-down, then

>    click :guilabel:`Turnstile`.

Reviewer: I think we can consider everything in that sidebar to be a menu:

```suggestion
#. In the dashboard's navigation sidebar, go to :menuselection:`Application security --> Turnstile`.
```

#. On the :guilabel:`Overview` page, click :icon:`fa-plus` :guilabel:`Add widget`.
#. Add a :guilabel:`Widget name` to easily identify it.
#. Click :icon:`fa-plus` :guilabel:`Add Hostnames`, enter a custom hostname (e.g., *example.com* or
   *subdomain.example.com*), then click :guilabel:`Add` twice.
#. Select a :guilabel:`Widget Mode`:

   - The :guilabel:`Managed` mode is **recommended**, as visitors can be prompted to check a box

>      confirming they are human if Turnstile deems necessary.

Reviewer: ```suggestion
   - The :guilabel:`Managed` mode is **recommended**, as it allows Turnstile to prompt visitors to confirm they are human when necessary.
```

     .. image:: spam_protection/turnstile-human.png
        :alt: Cloudflare Turnstile human verification widget

   - For the :guilabel:`Non-interactive` and :guilabel:`Invisible` modes, visitors are never
     prompted to interact. In :guilabel:`Non-interactive` mode, a loading widget can be displayed to
     warn visitors that Turnstile protects the form; however, the widget is not supported by Odoo.

     .. note::
        If the Turnstile check fails, visitors are not able to submit the form, and the following
        error message is displayed:

        .. image:: spam_protection/turnstile-error.png
           :alt: Cloudflare Turnstile verification error message

#. Click :guilabel:`Create`.

The generated keys are then displayed. Leave the page open for convenience, as copying the keys in
Odoo is required next.

On Odoo
-------

#. From the database dashboard, open the **Settings** app. Under :guilabel:`Integrations`, enable

>    :guilabel:`Cloudflare Turnstile` and click :guilabel:`Save`.

Reviewer: ```suggestion
   :guilabel:`Cloudflare Turnstile`, then click :guilabel:`Save`.
```
(nitpick! but since *Save* is not under *Integrations* ...)

> #. Open the Cloudflare Turnstile page, copy the :guilabel:`Site Key`, and paste it into the

Reviewer: I think I'd add a reference to the "On Cloudflare" section here?

   :guilabel:`CF Site Key` field in Odoo.
#. Open the Cloudflare Turnstile page, copy the :guilabel:`Secret Key`, and paste it into the
   :guilabel:`CF Secret Key` field in Odoo.
#. Click :guilabel:`Save`.

.. tip::

>    Navigate to :guilabel:`Turnstile` in your Cloudflare account to view analytics and access

Reviewer: Do we mean:

```suggestion
   Navigate to :menuselection:`Application security --> Turnstile` in your Cloudflare account to view analytics and access
```
? (i.e., the Overview page you mentionned earlier?)

> On the website registration page:

Reviewer: Following my comment above, I think i'd remove this sentence entirely

#. Enter a :guilabel:`Label` for the website, e.g., *example.com*.
#. Leave the :guilabel:`reCAPTCHA type` set to :guilabel:`Score based (v3)`.
#. Enter one or more :guilabel:`Domains` (e.g., *example.com* or *subdomain.example.com*).
#. Under :guilabel:`Google Cloud Platform`, a project is automatically created or selected if one
   already exists for the logged-in Google account. Click the field to select a project manually
   or rename the automatically created project.

> #. Click :guilabel:`Submit`.

Reviewer: ```suggestion
#. Agree to the terms of service.
#. Click :guilabel:`Submit`.
```

<img width="1025" height="180" alt="Image" src="https://github.com/user-attachments/assets/f2cf8d52-70fe-4cdc-a46b-0066fcc62c6e" />

Author: @auva-odoo My screenshot is not correct because it doesn't show the terms of service box. Can you take a full screenshot of your recaptcha page (if you have not yet accepted the terms!?). Or should we just remove the screenshot? I am not sure if it is that valuable either

Reviewer: I think you're right, we can just remove it actually :slightly_smiling_face:

.. image:: spam_protection/recaptcha-google-configuration-page.png
   :scale: 70%
   :alt: reCAPTCHA website registration example

The generated keys are then displayed. Leave the page open for convenience, as copying the keys in
Odoo is required next.

On Odoo
-------

- From the database dashboard, open the **Settings** app. Under :guilabel:`Integrations`, navigate
  to the :guilabel:`reCAPTCHA` section.

  .. warning::
     Do not uninstall the :guilabel:`Google reCAPTCHA integration` module, as it would also remove
     many other modules.

- Open the Google reCAPTCHA page, click :guilabel:`COPY SITE KEY`, and paste it into the
  :guilabel:`Site Key` field in Odoo.
- Open the Google reCAPTCHA page, click :guilabel:`COPY SECRET KEY`, and paste it into the
  :guilabel:`Secret Key` field in Odoo.
- Change the default :guilabel:`Minimum score` (`0.70`) if necessary, using a value between `0.00`
  and `1.00`. The higher the threshold is, the harder it is to pass the reCAPTCHA, and vice
  versa.

> - Click :guilabel:`Save`.

Reviewer: Should this be a numbered list instead? 

--> if so, I think I'd add lines 110-111 in the steps as well (at the beginning of the list then!)

.. seealso::
   `Interpret reCAPTCHA scores - Google documentation <https://cloud.google.com/recaptcha/docs/interpret-assessment-website#interpret_scores>`_

> You can notify visitors that reCAPTCHA protects a form. To do so, navigate to the form and open the

Reviewer: Should we add a reference to [this section](https://www.odoo.com/documentation/master/applications/websites/website/web_design/building_blocks.html#form)?

website editor. Then, click somewhere on the form, go to the :guilabel:`Customize` tab, and, in

> the :guilabel:`Form` section, enable the :guilabel:`Show ReCAPTCHA Policy` option.

Reviewer: ```suggestion
the :guilabel:`Form` section, enable :guilabel:`Show ReCAPTCHA Policy`.
```
